core/src/main/java/org/apache/calcite/sql/SqlOperatorBinding.java

```diff
@@ -150,7 +150,7 @@ public Comparable getOperandLiteralValue(int ordinal) {
    * <p>This is only relevant for SQL validation.
    *
    * @param ordinal   zero-based ordinal of operand of interest
-   * @param allowCast whether to regard CAST(constant) as a constant
+   * @param allowCast whether to regard CAST(null) as null
    * @return whether operand is null; false for everything except SQL
```

Comment:
Why to change the comment, look at the implementation in `SqlUtil#isNullLiteral`, you have change the semantics to be wrong, please change it back.

Reply:
Ok